core/com.b2international.snowowl.datastore/src/com/b2international/snowowl/core/repository/TerminologyRepositoryConfigurer.java

```diff
@@ -32,4 +36,9 @@
 	 */
 	Collection<Class<?>> getAdditionalMappings();
 
+	/**
+	 * @return additional component deletion policies for subtypes of RevisionDocument
+	 */
+	Map<Class<?>, Predicate<RevisionDocument>> getComponentDeletionPolicies();
```

Comment:
This can also be hidden entirely behind the `ComponentDeletionPolicy` interface:

```suggestion
	ComponentDeletionPolicy getComponentDeletionPolicy();
```

Reply:
But we need that class information to be able to select the appropriate policy for a given type, don't we?

Reply:
My mistake; wanted to say `CompositeComponentDeletionPolicy`. This ties into my previous comment, this could be done if there would be a way to merge multiple composites into one, eg. via a `CompositeComponentDeletionPolicy#builder(CompositeComponentDeletionPolicy)` or `CompositeComponentDeletionPolicy.Builder#withPolicies(CompositeComponentDeletionPolicy)` method.